tests/unit/test_redshift_adapter.py

```diff
@@ -447,6 +447,23 @@ def test_invalid_iam_no_cluster_id(self):
 
         self.assertTrue("'cluster_id' must be provided" in context.exception.msg)
 
+    @mock.patch("redshift_connector.connect", MagicMock())
+    def test_connection_has_backend_pid(self):
```

Comment:
I think this test holds up even with a very different approach. That's a sign of a good test.